source/credentials_provider_imds.c

```diff
@@ -393,11 +486,37 @@ static int s_make_imds_http_query(
         goto on_error;
     }
 
-    if (aws_http_message_set_request_path(request, *uri)) {
-        goto on_error;
+    struct aws_credentials_provider_imds_impl *impl = imds_user_data->imds_provider->impl;
+
+    if (imds_user_data->query_state == AWS_IMDS_QS_TOKEN) {
```

Comment:
Feels like this new block could go in a separate token-oriented function.

Reply:
Yes, it's better